sdk-ext-lib/src/main/java/cash/z/ecc/sdk/model/ZecSend.kt

```diff
@@ -0,0 +1,47 @@
+package cash.z.ecc.sdk.model
+
+import kotlinx.coroutines.runBlocking
+
+data class ZecSend(val destination: WalletAddress, val amount: Zatoshi, val memo: Memo) {
+    companion object
+}
+
+fun ZecSend.Companion.new(
+    destinationString: String,
+    zecString: String,
+    memoString: String,
+    monetarySeparators: MonetarySeparators
+): ZecSendValidation {
+
+    // This runBlocking shouldn't have a performance impact, since everything needs to be loaded at this point.
+    // However it would be better to eliminate it entirely.
+    val destination = runBlocking { WalletAddress.Unified.new(destinationString) }
+    val amount = Zatoshi.fromZecString(zecString, monetarySeparators)
+    val memo = Memo(memoString)
+
+    val validationErrors = buildSet {
+        if (null == amount) {
+            add(ZecSendValidation.Invalid.ValidationError.INVALID_AMOUNT)
+        }
+
+        // TODO [#250]: Implement all validation
```

Comment:
Yeah, I don't know what the right answer will be yet either.  This code API lives in sdk-ext, which means that I hope we eventually move it into the SDK once we stabilize and figure out how this might be able to enhance the SDK.

This is my current thought process:

I think an ideal public API for the SDK is strongly typed.  The implementation I ended up with is a `ZecSend` object which is composed of `Address`, `Zatoshi`, and `Memo`.  Each of those is also strongly typed, verifying that they meet basic validity.  This contrasts to our current SDK public API which is basically string, long, and string, which I don't like as much.  (It makes it too easy to transpose the address and the memo fields, for example).

`ZecSend` isn't really doing that much validation itself.  It mostly just checks is the Zec amount enough to cover the transaction fee.  Everything else is delegated to the other objects, like `Address`, `Zatoshi`, and `Memo`.  Note that `Address` will delegate to the Rust code for its validation.

The main complexity here is converting those validations into some kind of enum or structure that the UI can then easily convert to display to the user.  Even if Rust code does the validation, the conversion into a set of enums probably needs to still live here.

Thoughts?